sway-ir/src/parser.rs

```diff
@@ -174,6 +178,26 @@ mod ir_builder {
                     IrAstOperation::BitCast(val, ty)
                 }
 
+            rule op_binary_add() -> IrAstOperation
+                = "add" _ arg1:id() comma() arg2:id() {
+                    IrAstOperation::BinaryOp(BinaryOpKind::Add, arg1, arg2)
+                }
+
+            rule op_binary_sub() -> IrAstOperation
+                = "sub" _ arg1:id() comma() arg2:id() {
+                    IrAstOperation::BinaryOp(BinaryOpKind::Sub, arg1, arg2)
+                }
+
+            rule op_binary_mul() -> IrAstOperation
+                = "mul" _ arg1:id() comma() arg2:id() {
+                    IrAstOperation::BinaryOp(BinaryOpKind::Mul, arg1, arg2)
+                }
+
+            rule op_binary_div() -> IrAstOperation
+                = "div" _ arg1:id() comma() arg2:id() {
+                    IrAstOperation::BinaryOp(BinaryOpKind::Div, arg1, arg2)
+                }
+
```

Comment:
This is a bit redundant.  Instead have a rule for `op_binary_kind` returning `BinaryOpKind` which matches the `"add"` / `"sub"` / etc. and then a single `op_binary` rule which calls it.